simapp/state.go

```diff
@@ -12,94 +10,75 @@ import (
 	"github.com/tendermint/tendermint/crypto/secp256k1"
 	tmtypes "github.com/tendermint/tendermint/types"
 
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types/module"
 	"github.com/cosmos/cosmos-sdk/x/genaccounts"
 	"github.com/cosmos/cosmos-sdk/x/simulation"
 )
 
 // AppStateFn returns the initial application state using a genesis or the simulation parameters.
 // It panics if the user provides files for both of them.
 // If a file is not given for the genesis or the sim params, it creates a randomized one.
-func AppStateFn(
-	r *rand.Rand, accs []simulation.Account, config simulation.Config,
-) (appState json.RawMessage, simAccs []simulation.Account, chainID string, genesisTimestamp time.Time) {
+func AppStateFn(cdc *codec.Codec, simManager *module.SimulationManager) simulation.AppStateFn {
+	return func(r *rand.Rand, accs []simulation.Account, config simulation.Config,
+	) (appState json.RawMessage, simAccs []simulation.Account, chainID string, genesisTimestamp time.Time) {
+
+		if flagGenesisTimeValue == 0 {
+			genesisTimestamp = simulation.RandTimestamp(r)
+		} else {
+			genesisTimestamp = time.Unix(flagGenesisTimeValue, 0)
+		}
 
-	cdc := MakeCodec()
```

Comment:
use the app codec as otherwise this would only work with types/interfaces registered by the SDK